src/Core/Form/IdentifiableObject/DataHandler/DiscountFormDataHandler.php

```diff
@@ -170,6 +173,20 @@ private function updateDiscountConditions(int $discountId, array $data): void
             }
         }
 
+        $specificProducts = $data['conditions']['cart_conditions']['specific_products'] ?? [];
```

Comment:
This part must be included in an if statement, based on the `$data['conditions']['cart_conditions']['children_selector']` selector value And only when cart conditions are selected also So it must be moved in the upper block with a new block:
```php
elseif ($data['conditions']['cart_conditions']['children_selector'] === CartConditionsType::SPECIFIC_PRODUCTS) {
```